client/src/components/UsersTable.js

```diff
@@ -53,8 +72,7 @@ export default function UsersTable({ useAuthHandler, onlyStudents, onlyProfessor
                                         <th><Link to={`/user/${j.username}`}>{j.username}</Link></th>
                                         <th>{j.surname}</th>
                                         <th>{j.name}</th>
-                                        { !onlyProfessors && currentUser && currentUser.is_professor &&
-                                        <th><Button size="xs" color="danger" onClick={async () => await deleteUser(j.id)}>Remove</Button></th> }
+                                        { renderRemoveButton() }
```

Comment:
You need to call the function with the student id in order to make the deletion work
```suggestion
                                        { renderRemoveButton(j.id) }
```

Reply:
good catch 👍